openassessment/staffgrader/staff_grader_mixin.py

```diff
@@ -206,53 +244,35 @@ def _get_list_workflows_serializer_context(self, staff_workflows, is_team_assign
         course_id = self.get_student_item_dict()['course_id']
 
-        # Fetch user identifier mappings
+        context = {}
+
         if is_team_assignment:
             # Look up the team IDs for submissions so we can later map to team names
             team_submission_uuid_to_team_id = get_team_ids_by_team_submission_uuid(submission_uuids)
 
             # Look up names for teams
             topic_id = self.selected_teamset_id
             team_id_to_team_name = self.teams_service.get_team_names(course_id, topic_id)
-
-            # Do bulk lookup for scorer anonymous ids (submitting team name is a separate lookup)
-            anonymous_id_to_username = map_anonymized_ids_to_usernames(
-                set(workflow_scorer_ids)
-            )
-            anonymous_id_to_email = map_anonymized_ids_to_emails(
-                set(workflow_scorer_ids)
-            )
-            anonymous_id_to_fullname = map_anonymized_ids_to_fullname(
-                set(workflow_scorer_ids)
-            )
-
-            context = {
+            context.update({
                 'team_submission_uuid_to_team_id': team_submission_uuid_to_team_id,
                 'team_id_to_team_name': team_id_to_team_name,
-            }
+            })
         else:
             # When we look up usernames we want to include all connected learner student ids
             submission_uuid_to_student_id = get_student_ids_by_submission_uuid(
                 course_id,
                 submission_uuids,
             )
+            context['submission_uuid_to_student_id'] = submission_uuid_to_student_id
 
-            # Do bulk lookup for all anonymous ids (submitters and scoreres). This is used for the
-            # `gradedBy` and `username` fields
-            anonymous_id_to_username = map_anonymized_ids_to_usernames(
-                set(submission_uuid_to_student_id.values()) | workflow_scorer_ids
-            )
-            anonymous_id_to_email = map_anonymized_ids_to_emails(
-                set(submission_uuid_to_student_id.values()) | workflow_scorer_ids
-            )
-
-            anonymous_id_to_fullname = map_anonymized_ids_to_fullname(
-                set(submission_uuid_to_student_id.values()) | workflow_scorer_ids
-            )
+        all_anonymous_ids = set(workflow_scorer_ids)
+        if not is_team_assignment:
+            all_anonymous_ids |= set(context['submission_uuid_to_student_id'].values())
 
-            context = {
-                'submission_uuid_to_student_id': submission_uuid_to_student_id,
-            }
+        anonymous_id_to_user_data = map_anonymized_ids_to_user_data(all_anonymous_ids)
+        anonymous_id_to_username = {key: val["username"] for key, val in anonymous_id_to_user_data.items()}
+        anonymous_id_to_email = {key: val["email"] for key, val in anonymous_id_to_user_data.items()}
+        anonymous_id_to_fullname = {key: val["fullname"] for key, val in anonymous_id_to_user_data.items()}
```

Comment:
why is this outside the else now?

Reply:
Beacuse it was the same code block repeated twice inside the `If..Else` statement... so, with the refactor in wich I merge `anonymous_id_to_username`, `anonymous_id_to_email` and `anonymous_id_to_fullname` into `anonymous_id_to_user_data` , I was forced to modify this function that was relative to the first endpoint and improved it a little.